File: modules/angular2/test/tools/tools_spec.ts
import {
  afterEach,
  beforeEach,
  ddescribe,
  describe,
  expect,
  iit,
  inject,
  it,
  xit
} from 'angular2/test_lib';

import {enableDebugTools, disableDebugTools} from 'angular2/tools';
import {SpyComponentRef, callNgProfilerTimeChangeDetection} from './spies';

export function main() {
  describe('profiler', () => {
    beforeEach(() => { enableDebugTools((<any>new SpyComponentRef())); });

    afterEach(() => { disableDebugTools(); });

    it('should time change detection', () => { callNgProfilerTimeChangeDetection(); });

    it('should time change detection with recording',
       () => { callNgProfilerTimeChangeDetection({'record': true}); });
  });
}

File: test/tools/tools_spec.dart
library angular2.test.tools.tools_spec;

import "package:angular2/testing_internal.dart"
    show
        afterEach,
        beforeEach,
        ddescribe,
        describe,
        expect,
        iit,
        inject,
        it,
        xit;
import "package:angular2/tools.dart" show enableDebugTools, disableDebugTools;
import "spies.dart" show SpyComponentRef, callNgProfilerTimeChangeDetection;

main() {
  describe("profiler", () {
    beforeEach(() {
      enableDebugTools(((new SpyComponentRef() as dynamic)));
    });
    afterEach(() {
      disableDebugTools();
    });
    it("should time change detection", () {
      callNgProfilerTimeChangeDetection();
    });
    it("should time change detection with recording", () {
      callNgProfilerTimeChangeDetection({"record": true});
    });
  });
}
